src/nodes/lighting/EnvironmentNode.js

```diff
@@ -25,17 +25,21 @@ class EnvironmentNode extends LightingNode {
 
 	setup( builder ) {
 
+		const { material } = builder;
+
 		let envNode = this.envNode;
 
-		if ( envNode.isTextureNode ) {
+		if ( envNode.isTextureNode || envNode.isMaterialReferenceNode ) {
```

Comment:
Exchanging the env map for basic, lambert and phong works now as expected. Great!

However, `EnvironmentNode` needs an enhancement since the PMREM isn't regenerated when `envMap` is changed. 

`updateBefore()` might be a good spot for checking the status but I'm not sure how to access the material without the builder reference. Besides, ideally we only update the PMREM node without regenerating the `radiance` and `irradiance` node.

Reply:
It seems like having a `reference()/materialReference()` supporting `pmrem` we could be removed this cacheEnv using `MeshStandardNodeMaterial.setupEnvironment()` for create this.

Reply:
We need the regeneration logic for `Scene.background` as well so it makes sense to move the PMREM generation to a more central place.

Reply:
Let's tackle this in a different PR.